File: src/main/java/de/bst/five_cities/utils/SparkUtils.java
package de.bst.five_cities.utils;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;

public class SparkUtils {

	private static JavaSparkContext sparkContext;

	public static JavaSparkContext getSparkContext() {
		if (sparkContext != null)
			return sparkContext;

		SparkConf sparkConf = new SparkConf();
		sparkConf.setAppName("Hallo Spark!");
		sparkConf.setMaster("local[*]");

		sparkContext = new JavaSparkContext(sparkConf);
		return sparkContext;
	}

	public static void close() {
		if (sparkContext != null) {
			sparkContext.close();
			sparkContext = null;
		}
	}

	public static <T> T debugPrint(T something) {
		System.out.println(">> " + something);
		return something;
	}

}

File: src/main/scala/de/bst/five_cities/utils/SparkUtils.scala
package de.bst.five_cities.utils

import org.apache.spark.{SparkConf, SparkContext}

object SparkUtils {
  private var sparkContext: Option[SparkContext] = None

  def getSparkContext: SparkContext = {
    sparkContext getOrElse {
      val sparkConf: SparkConf = new SparkConf
      sparkConf.setAppName("Hallo Spark!")
      sparkConf.setMaster("local[*]")
      sparkContext = Some(new SparkContext(sparkConf))
      sparkContext.get
    }
  }

  def close {
    sparkContext foreach (_.stop)
    sparkContext = None
  }

  def debugPrint[T](something: T): T = {
    println(s">> $something")
    something
  }
}
